babble/src/main/java/io/mosaicnetworks/babble/servicediscovery/webrtc/WebRTCService.java

```diff
@@ -208,6 +253,44 @@ public void onErrorResponse(VolleyError error) {
 
     @Override
     public void stopAdvertising() {
-        //do nothing
+        Disco group = ConfigManager.getInstance(null).getDisco();
+
+        // TODO: this is a temporary hack to prevent anyone other than the group creator to delete
+        // the group from disco. The functionality should be propertly implemented server side.
+        if (!group.PubKey.equals(ConfigManager.getInstance(null).getPublicKey())) {
+            Log.d(TAG, "Not group creator => not deleting from disco");
```

Comment:
Again this feels wrong with the assumption that ConfigManager persists. Fortunately there is an open pull request moving getting the public key to the service. 

https://github.com/mosaicnetworks/babble-android/pull/67